themes-default/slim/views/vue-components/app-link.mako

```diff
@@ -22,80 +23,109 @@ Vue.component('app-link', {
         placeholder: {
             type: String,
             default: 'indexer-to-name'
-        }
+        },
+        base: String
     },
     computed: {
+        config() {
+            return this.$store.state.config;
+        },
+        computedBase() {
+            // Return prop before HTML element to allow tests to mock
+            if (this.base) {
+                return this.base;
+            }
+
+            return document.getElementsByTagName('base')[0].getAttribute('href');
```

Comment:
I don't see any problem with doing this:
```js
const { base } = this;
// Return prop before HTML element to allow tests to mock
return this.base || document.getElementsByTagName('base')[0].getAttribute('href');
```
It's just a personal preference.

Reply:
We need to make sure document isn't ever called since it'll throw an error in the tests. I know how bad that sounds but I'll fix it once I move this to a `.vue` file.

Reply:
AFAIK, JavaScript logical statements are evaluated from left to right and not all at once.
Meaning if `Boolean(this.base)` is true, it will return that. As long as you pass the base in the tests, should be good.

Reply:
It's not that. It's the testing framework seeing it in the AST that's the issue.